File: scripts/inspect-repacked-qgz.ts
import * as fs from 'fs';
import * as path from 'path';
import * as admZip from 'adm-zip';

const qgzPath = 'QGIS Project Template/QGIS.qgz';
if (!fs.existsSync(qgzPath)) {
  console.error(`Error: ${qgzPath} not found.`);
  process.exit(1);
}

try {
  const zip = new admZip(qgzPath);
  const zipEntries = zip.getEntries();
  const qgsEntry = zipEntries.find(entry => entry.entryName.endsWith('.qgs'));
  
  if (!qgsEntry) {
    console.error('QGS file not found inside QGZ.');
    process.exit(1);
  }
  
  const content = qgsEntry.getData().toString('utf8');
  
  // Simple XML parsing/regex to find layers and datasources
  const maplayerRegex = /<maplayer[^>]*>([\s\S]*?)<\/maplayer>/g;
  let match;
  while ((match = maplayerRegex.exec(content)) !== null) {
    const layerXml = match[1];
    const nameMatch = /<layername>([^<]*)<\/layername>/.exec(layerXml);
    if (nameMatch && nameMatch[1].startsWith('SLT_')) {
      const name = nameMatch[1];
      const datasourceMatch = /<datasource>([^<]*)<\/datasource>/.exec(layerXml);
      const providerMatch = /<provider>([^<]*)<\/provider>/.exec(layerXml);
      console.log(`Layer: ${name}`);
      console.log(`  - Provider: ${providerMatch ? providerMatch[1] : 'None'}`);
      console.log(`  - Datasource: ${datasourceMatch ? datasourceMatch[1] : 'None'}`);
    }
  }
} catch (e: any) {
  console.error('Error inspecting zip:', e.message || e);
}

File: scripts/inspect-repacked-qgz.py
import zipfile
import xml.etree.ElementTree as ET
import os

qgz_path = 'QGIS Project Template/QGIS.qgz'
with zipfile.ZipFile(qgz_path, 'r') as zip_ref:
    qgs_file = [f for f in zip_ref.namelist() if f.endswith('.qgs')][0]
    with zip_ref.open(qgs_file) as xml_file:
        tree = ET.parse(xml_file)
        root = tree.getroot()
        
        for layer in root.findall('.//maplayer'):
            name_elem = layer.find('layername')
            if name_elem is not None and name_elem.text.startswith('SLT_'):
                name = name_elem.text
                datasource = layer.find('datasource')
                provider = layer.find('provider')
                print(f"Layer: {name}")
                print(f"  - Provider: {provider.text if provider is not None else 'None'}")
                print(f"  - Datasource: {datasource.text if datasource is not None else 'None'}")
